### PR TITLE
Added Atlas deployment guide and sidebar link

### PR DESCRIPTION
**Pull request by TeresaGobble**
## Description

This PR introduces the Atlas deployment guide we had in our [Legacy documents](https://legacy.faustjs.org/docs/going-to-production/deployment) for Faustjs.org.

Please double check the Node versioning in the first paragraph: `Faust.js supports Node.js v14.x and v16.x. Please make sure you are using a Node.js version that is compatible when deploying to avoid unexpected errors.`

### CODE REVIEW

File: internal/faustjs.org/docs/going-to-production/deployment.mdx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Building Your App
+
+Faust.js is build on top of [Next.js](https://nextjs.org/), so the build process is identical. Take a look at the [Next.js Build API](https://nextjs.org/docs/deployment#nextjs-build-api) for info relating to how Next.js generates an optimized production build.
+
+## Deploying To Atlas
+
+Atlas is the most effortless way to deploy a Faust.js app. Connect your GitHub repo and Atlas will automatically build and deploy your Faust.js project on each push. Some of the benefits you get out of the box are:
+
+- Automatic installation of required plugins ([FaustWP](https://wordpress.org/plugins/faustwp/), [WPGraphQL](https://wordpress.org/plugins/wp-graphql/), etc.)
```
Comment (josephfusco) [inline internal/faustjs.org/docs/going-to-production/deployment.mdx:18]:
```suggestion
- Automatic installation of required WordPress plugins ([Faust](https://wordpress.org/plugins/faustwp/), [WPGraphQL](https://wordpress.org/plugins/wp-graphql/), etc.)
```

File: internal/faustjs.org/docs/going-to-production/deployment.mdx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- Automatic setup of Faust.js environment variables
+- Automatic configuration of FaustWP headless settings
```
Comment (josephfusco) [inline internal/faustjs.org/docs/going-to-production/deployment.mdx:20]:
```suggestion
- Automatic configuration of the Faust WordPress plugin's settings
```

File: internal/faustjs.org/sidebars.js [hunk 3]
```diff
@@ -101,9 +101,14 @@ module.exports = {
     },
     {
       type: 'doc',
-      label: 'Migration from old Faust',
+      label: 'Migration from Old Faust',
```
Comment (josephfusco) [inline internal/faustjs.org/sidebars.js:104]:
```suggestion
      label: 'Migration from legacy Faust',
```

**Review by josephfusco — CHANGES_REQUESTED** [review]
Great work! Left some minor suggestions

**Comment by TeresaGobble** [comment]
> Great work! Left some minor suggestions

Awesome catches, Joe, thank you so much!

**Review by josephfusco — APPROVED** [review]
🔥

**Pull request merged**